src/core/dev_api/shape_util.hpp

```diff
@@ -25,6 +25,6 @@ Shape make_dynamic_shape();
  * @return True if shape is dynamic otherwise false.
  */
 OPENVINO_DEPRECATED("This function is deprecated and will be removed soon.")
-bool is_dynamic_shape(const Shape& s);
+OPENVINO_API bool is_dynamic_shape(const Shape& s);
```

Comment:
can you check if export of this required. I think is used in core only. 
If not required maybe is better not export as also marked as deprecated.

Reply:
It is used by the CPU plugin as well, so this is the reason why they are exported. I also though about making these utils header only, but the usage of a static variable changed my mind.